### PR TITLE
Feature(state-view): TokenInfo types and max supply

### LINKED ISSUE #168: State View
**Description:**
- Return newly added properties in Token Get Info queries

### LINKED ISSUE #169: Fees
**Description:**
- Update fee calculation based on the new properties

### LINKED ISSUE #170: E2E Tests
**Description:**
- Update existing E2E tests (fungible)
- Add new E2E tests (Non fungible type)

### PR DESCRIPTION
**Pull request by failfmi**
**Related issue(s)**:
Closes #168 
Closes #169 
Closes #170 

**Summary of the change**:
* `TokenInfo`: Set types and max supply
* E2E: assert valid set types

**External impacts**:
None.

**Applicable documentation**
- [ ] Javadoc
- [ ] HAPI protobuf comments
- [ ] README

### CODE REVIEW

**Review by Daniel-K-Ivanov — APPROVED** [review]

**Review by joan41868 — APPROVED** [review]

File: hedera-node/src/main/java/com/hedera/services/context/primitives/StateView.java [hunk 1]
```diff
@@ -237,13 +237,15 @@ public StateView(
 			}
 			var token = tokenStore.get(id);
 			var info = TokenInfo.newBuilder()
```
Comment (Daniel-K-Ivanov) [inline hedera-node/src/main/java/com/hedera/services/context/primitives/StateView.java:239]: 1) We will have to add `supplyType` as-well
2) StateViewTest.java should be updated with checks for the new properties as-well

File: test-clients/src/main/java/com/hedera/services/bdd/spec/queries/token/HapiGetTokenInfo.java [hunk 2]
```diff
@@ -83,78 +86,106 @@ protected HapiGetTokenInfo self() {
 		return this;
 	}
 
+	public HapiGetTokenInfo hasTokenType(TokenType t) {
```
Comment (Daniel-K-Ivanov) [inline test-clients/src/main/java/com/hedera/services/bdd/spec/queries/token/HapiGetTokenInfo.java:89]: we should add `hasSupplyType` method as-well

File: test-clients/src/main/java/com/hedera/services/bdd/spec/queries/token/HapiGetTokenInfo.java [hunk 3]
```diff
@@ -164,6 +195,12 @@ public HapiGetTokenInfo isNotDeleted() {
 	protected void assertExpectationsGiven(HapiApiSpec spec) throws Throwable {
 		var actualInfo = response.getTokenGetInfo().getTokenInfo();
```
Comment (Daniel-K-Ivanov) [inline test-clients/src/main/java/com/hedera/services/bdd/spec/queries/token/HapiGetTokenInfo.java:196]: we should assert `tokenSupplyType` as-well

File: test-clients/src/main/java/com/hedera/services/bdd/suites/token/TokenCreateSpecs.java [hunk 4]
```diff
@@ -201,14 +201,20 @@ public HapiApiSpec creationHappyPath() {
 								.treasury(TOKEN_TREASURY)
 								.autoRenewAccount("autoRenewAccount")
 								.autoRenewPeriod(A_HUNDRED_SECONDS)
+								.maxSupply(1000)
```
Comment (Daniel-K-Ivanov) [inline test-clients/src/main/java/com/hedera/services/bdd/suites/token/TokenCreateSpecs.java:204]: Specifying only `maxSupply` without `supplyType` `FINITE` should result in an error

**Review by Daniel-K-Ivanov — CHANGES_REQUESTED** [review]
The following BDD tests could be implemented:
1) Token's supply type is correctly returned on `GetTokenInfo` queries - FINITE for `maxSupply` != 0 and `INFINITE` for `maxSupply=0`
2) Tokens cannot be created if `supply` type is `FINITE`, but `maxSupply` is `0`
3) Tokens cannot be created if `supply` type is `INFINITE`, but `maxSupply` is !=0

**Review by Daniel-K-Ivanov — APPROVED** [review]

**Pull request merged**